src/main/java/seedu/edrecord/model/assignment/Assignment.java

```diff
@@ -50,7 +59,7 @@ public boolean isSameAssignment(Assignment otherAssignment) {
             return true;
         }
         return otherAssignment != null
-                && otherAssignment.getName().equals(getName());
```

Comment:
Do you think it's a good idea to add ID check into `isSameAssignment`? Assignment name and ID should be both unique anyway, a lot of code is checking for duplicate name and/or ID so we can just put both under 1 method.

Reply:
I think is better to split because in case the user change the id but not the name in the JSON file then need check both anyway